### PR TITLE
ceph-volume: detect ceph-disk osd if PARTLABEL is missing

### PR DESCRIPTION
**Pull request by jan--f**
Fixes: http://tracker.ceph.com/issues/40917

The issue is that ceph-disk seems to sometimes not assign a PARTLABEL for wal and db partitions. This PR adds a fall back to detect ceph-disk OSDs by PARTTYPE. I pulled the respective UUIDs out of the luminous ceph-disk code. I so far only included the `regular` UUIDs. It is however straight forward to add others as well, @alfredodeza what do you think?

This also restructures some test fixtures to improve code re-use and make it easier to add new tests.

- [x] References tracker ticket
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by jan--f** [comment]
cc @jschmid1

**Comment by jan--f** [comment]
jenkins test ceph-volume tox

**Comment by jan--f** [comment]
jenkins test ceph-volume simple all

**Comment by alfredodeza** [comment]
It is troubling that ceph-disk had different behavior for setting these in Bluestore. Perhaps it is sgdisk not setting them correctly. In any case, I think that falling back to another method when failing to detect it is reasonable

File: src/ceph-volume/ceph_volume/util/device.py [hunk 1]
```diff
@@ -9,6 +9,15 @@
 report_template = """
 {dev:<25} {size:<12} {rot!s:<7} {available!s:<9} {model}"""
 
+ceph_disk_parttypes = {
```
Comment (alfredodeza) [inline src/ceph-volume/ceph_volume/util/device.py]: even though these seemed to be used here only, the ceph-disk GUIDs are in `util/constants.py` so perhaps getting these over there might be nice for consistency

Comment (jan--f) [reply src/ceph-volume/ceph_volume/util/device.py]: Ah thx, yes I'll use those. Two questions though:

1. The plain type has `'encrypted': True, 'encryption_type': 'plain'`. Is that intentional or should that be `'encrypted': False, 'encryption_type': 'plain'`?
2. The lockbox label is missing. Probably no harm in adding that so we can detect it in inventory right?

**Comment by jan--f** [comment]
jenkins test ceph-volume tox

File: src/ceph-volume/ceph_volume/util/device.py [hunk 2]
```diff
@@ -416,12 +417,24 @@ def partlabel(self):
             return lsblk_partlabel
         return self.device.blkid_api.get('PARTLABEL', '')
 
+    @property
+    def parttype(self):
+        """
+        Seems like older version do not detect PARTTYPE correctly (assuming the
+        info in util/disk.py#lsblk is still valid).
+        SImply resolve to using blkid since lsblk will throw an error if asked
```
Comment (jschmid1) [inline src/ceph-volume/ceph_volume/util/device.py]: probably fine for this fix here, but we should look in `lsblk` as to why it raises an error when asked for an unknown column

Comment (jan--f) [reply src/ceph-volume/ceph_volume/util/device.py]: I think this is just how lsblk is implemented.

File: src/ceph-volume/ceph_volume/util/device.py (hunk 1 above)
Comment (alfredodeza) [reply src/ceph-volume/ceph_volume/util/device.py]: 1) Great question. This is part of the types of encryption dmcrypt offers, one of them is `plain` (see [some docs about this here](https://wiki.archlinux.org/index.php/Dm-crypt/Device_encryption#Encrypting_devices_with_plain_mode)) and the other one is `luks`. So `plain` *is* a valid form of encryption.

2) Yes, lets add that, might as well try to get those in... who knows if we might need them later. Good catch!

Comment (jan--f) [reply src/ceph-volume/ceph_volume/util/device.py]: K, this should do the trick now.

File: src/ceph-volume/ceph_volume/util/constants.py [hunk 3]
```diff
@@ -23,11 +23,15 @@
...
     '45b0969e-9b03-4f30-b4c6-5ec00ceff106': {'type': 'journal', 'encrypted': True, 'encryption_type': 'plain'},
     'cafecafe-9b03-4f30-b4c6-5ec00ceff106': {'type': 'block', 'encrypted': True, 'encryption_type': 'plain'},
     '93b0052d-02d9-4d8a-a43b-33a3ee4dfbc3': {'type': 'block.db', 'encrypted': True, 'encryption_type': 'plain'},
     '306e8683-4fe2-4330-b7c0-00a917c16966': {'type': 'block.wal', 'encrypted': True, 'encryption_type': 'plain'},
-    '4fbd7e29-9d25-41b8-afd0-5ec00ceff05d': {'type': 'osd', 'encrypted': True, 'encryption_type': 'plain'},
+    '4fbd7e29-9d25-41b8-afd0-5ec00ceff05d': {'type': 'data', 'encrypted': True, 'encryption_type': 'plain'},
+    # regular guids that differ from plain
+    'fb3aabf9-d25f-47cc-bf5e-721d1816496b': {'type': 'lockbox', 'encrypted': False, 'encryption_type': None},
+    '30cd0809-c2b2-499c-8879-2d6b78529876': {'type': 'block.db', 'encrypted': False, 'encryption_type': None},
+    '5ce17fce-4087-4169-b7ff-056cc58473f9': {'type': 'block.wal', 'encrypted': False, 'encryption_type': None},
```
Comment (alfredodeza) [inline src/ceph-volume/ceph_volume/util/constants.py]: I double checked, and I believe that Wikipedia holds a good reference here for these GUIDs, might be worth adding the link as a comment (or if you want to double check as well) https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_type_GUIDs

Comment (jan--f) [reply src/ceph-volume/ceph_volume/util/constants.py]: ok I just added them all. This madness might as well live on.

**Pull request merged**